### ISSUE TITLE
Magisk support

### ISSUE THREAD

**Issue by Gooberpatrol66** [openandroidinstaller-dev/openandroidinstaller#92]
Can this tool be used to more easily install Magisk on top of LineageOS?

**Comment by tsterbak** [comment]
Hey, thanks for your interest in the project! :)
I have only very limited experience with Magisk since I usually don't root and I'm not super aware of the installation process. Can Magisk be sideloaded as a flashable zip or need some other things need to be patched?

**Comment by SirRGB** [comment]
yep, most people flash the apk or rename it to .zip and flash it (in case the twrp version is too old)
the recommended method is to patch the boot.img using the app, but no one really does that.

**Comment by anon1892** [comment]
Yes, flashing Magisk zip as "addon" works (used OrangeFox on lavender, see #218 and #222)

**Comment by MagicLike** [comment]
I used the "app patches boot.img" way and it also worked fine...

@anon1892 Does the addon flashing also work for twrp or only for orangefox?

**Comment by anon1892** [comment]
zip flashing is not specific to OrangeFox (which is based on TWRP), so it should work

**Comment by Doctorlao** [comment]
操作不来啊蛋蛋

**Comment by SirRGB** [comment]
The first step is done and will be included in 0.52. Flashing magisk as addon still requires renaming the .apk to .zip or simply ignoring the .zip suffix when selecting the file.

https://github.com/openandroidinstaller-dev/openandroidinstaller/commit/6859a5ebb299aac3e546823a4f2967ee2e56bd31

**Comment by tim-gromeyer** [comment]
Is there any progress on this? I installed LineageOS 20 on my Galaxy S6 yesterday (SM-G920F) and now I want to root it

**Comment by tsterbak** [comment]
I cannot guarantee that it works, since I did not test it. But if you sideload magisk as a zip addon, people reported that it works.

**Comment by tim-gromeyer** [comment]
Yes, that worked. Can we somehow automate this process into OpenAndroidInstaller?

**Comment by tsterbak** [comment]
> Yes, that worked. Can we somehow automate this process into OpenAndroidInstaller?

Great that it worked!

I'm not sure if I want to automate it more at the moment since I have no experience with Magisk. Are there official distributions that work on every device? Then we could think about integrating this. Otherwise, the best thing would be to have a short tutorial on how do do it in the repo or in the installer, if you want to use Magisk and root. What do you think?

**Comment by SirRGB** [comment]
Imo Magisk shouldnt be installable via a checkbox in the oai, because some devices dont handle it very well and might fail to boot. Also Magisk is aimed toward a little more experienced users, which I dont think this tool is aimed for, since you can shoot urself into ur own foot easily and it will cause some security related apps to fail if not hidden properly, tho that also depends on the rom itself...
